src/Ledger/TokenAlgebra/ValueSet.lagda

```diff
@@ -0,0 +1,221 @@
+  ι-identity .proj₁ tm {aa} = begin
+    lookup (ι ⊕ tm) aa          ≡⟨ ⊕-lemma ι tm ⟩
+    lookup ι aa + lookup tm aa  ≡⟨ cong (λ x → x + lookup tm aa) ι-lem3 ⟩
+    lookup tm aa                ∎
+  ι-identity .proj₂ tm {aa} = begin
+    lookup (tm ⊕ ι) aa          ≡⟨ ⊕-lemma tm ι ⟩
+    lookup tm aa + lookup ι aa  ≡⟨ cong (λ x → lookup tm aa + x) ι-lem3 ⟩
+    lookup tm aa + 0            ≡⟨ +-identityʳ (lookup tm aa) ⟩
+    lookup tm aa                ∎
+
+  open IsSemigroup
+  open IsMagma
+  isSemigrp : IsSemigroup _≋_ _⊕_
+  isSemigrp .isMagma .isEquivalence        = ≋-isEquivalence
+  isSemigrp .isMagma .∙-cong {u}{v}{x}{y}  = ⊕-cong {u}{v}{x}{y} -- (broken inference)
+  isSemigrp .assoc                         = ⊕-assoc
+
+  open IsMonoid
+  ≋-⊕-ι-isMonoid : IsMonoid _≋_ _⊕_ ι
+  ≋-⊕-ι-isMonoid .isSemigroup  = isSemigrp
+  ≋-⊕-ι-isMonoid .identity     = ι-identity
+\end{code}
+
+We are now in a position to define the commutative monoid.
+
+\begin{code}
+  open IsCommutativeMonoid
+  Vcm : CommutativeMonoid 0ℓ 0ℓ
+  Vcm .Carrier                        = AssetId ⇒ Quantity
+  Vcm ._≈_                            = _≋_
+  Vcm ._⋆_                            = _⊕_
+  Vcm .ε                              = ι
+  Vcm .isCommutativeMonoid .isMonoid  = ≋-⊕-ι-isMonoid
+  Vcm .isCommutativeMonoid .comm      = ⊕-comm
+
+  Value-TokenAlgebra :  (specialPolicy : PolicyId)
+                        (specialAsset : AssetName)
+                        (policies : AssetId ⇒ Quantity → ℙ PolicyId)
+                        (size : AssetId ⇒ Quantity → ℕ)
+                        {decTot : DecEq (AssetId ⇒ Quantity)}
+                        ⦃ dec : DecEq (PolicyId × AssetName) ⦄
+                        ---------------------------------------------
+                        → TokenAlgebra
+
+  Value-TokenAlgebra specialPolicy specialAsset policies size {decTot} ⦃ dec ⦄ =
+    record
+      { Value-CommutativeMonoid   = Vcm
+      ; coin                      = totalMap↠coin
+      ; inject                    = coin↪totalMap
+      ; policies                  = policies
+      ; size                      = size
+      ; _≤ᵗ_                      = leq
+      ; AssetName                 = AssetName
+      ; specialAsset              = specialAsset
+      ; property                  = compose-to-id
+      ; coinIsMonoidHomomorphism  = CoinMonHom
+      ; DecEq-Value               = decTot
+      }
+    where
+
+    specId : AssetId
+    specId = (specialPolicy , specialAsset)
+
+    open Update ⦃ dec ⦄
+
+    totalMap↠coin : AssetId ⇒ Quantity → Coin
+    totalMap↠coin tm = lookup tm specId
+
+    coin↪totalMap : Coin → AssetId ⇒ Quantity
+    coin↪totalMap c = update specId c zeroMap
+
+    leq : AssetId ⇒ Quantity → AssetId ⇒ Quantity → Type
+    leq = λ u v → ∀ {a}{p}{q} → lookup u (a , p) ≤ lookup v (a , q)
+
+    compose-to-id : totalMap↠coin ∘ coin↪totalMap ≗ id
+    compose-to-id _ = lookup-update-id zeroMap where open LookupUpdate {X = X}{specId}{⋁A}
```

Comment:
```suggestion
    compose-to-id _ = lookup-update-id zeroMap where open LookupUpdate {X = X} {specId} {⋁A}
```
Also, the `where` block should be on a new line, or use `let` if you want to do it in the same line instead.